pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -230,7 +227,7 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 	for _, node := range nodes {
 		fScore := float64(0)
 		if (maxCount - minCount) > 0 {
-			fScore = float64(schedulerapi.MaxPriority) * ((pm.counts[node.Name] - minCount) / (maxCount - minCount))
+			fScore = float64(schedulerapi.MaxPriority) * (float64(*pm.counts[node.Name]-minCount) / float64(maxCount-minCount))
```

Comment:
I guess, we might have lost the float scores by now. I believe, we tried something similar in the past and it affected correctness. The one thing, I am curious about is - any difference between the scores computed, is there some difference between the scores computed with this patch/without this patch?(IIRC, last time the issue was related to too many nodes having same score).

Reply:
@ravisantoshgudimetla could you point me the issue #?

This PR introduced following changes related with fraction:

- `weight` is always an integer, so new change that loads `weight` into `p.counts[node.Name]` (from float64 to int64) in an atomic way should be fine.
- change from `float64(term.Weight*int32(multiplier)))` to `int64(term.Weight*int32(multiplier))` should be also fine as `multiplier` is of type `int`
- `maxCount/minCount` are assigned from `p.counts[node.Name]`; should be good as well
- the last one: change from `(pm.counts[node.Name] - minCount) / (maxCount - minCount)` to `float64(*pm.counts[node.Name]-minCount) / float64(maxCount-minCount)`; this doens't lose correctness either.

Reply:
Got it, thanks for the explanation. I have some questions on the `term.Weight*int32(multiplier)` but I see that both of them are of the type int.